### PR TITLE
Add null pointer checks for improved error handling

### PR DESCRIPTION
**Pull request by Russsnizza**
Fixes: https://github.com/perfectsense/gyro-aws-provider/issues/671

## Description
This PR introduces necessary null pointer checks across several methods to prevent potential `NullPointerExceptions`.

### CODE REVIEW

File: src/main/java/gyro/aws/rds/DbClusterResource.java [hunk 1]
```diff
@@ -963,11 +964,12 @@ protected void doCreate(GyroUI ui, State state) {
...
                     .iops(getIops())
                     .manageMasterUserPassword(getManageMasterUserPassword())
-                    .masterUserSecretKmsKeyId(getMasterUserSecretKmsKey().getId())
+                    .masterUserSecretKmsKeyId(
+                        getMasterUserSecretKmsKey() == null ? null : getMasterUserSecretKmsKey().getId())
                     .serverlessV2ScalingConfiguration(getServerlessV2ScalingConfiguration() != null ?
                         getServerlessV2ScalingConfiguration().toServerlessV2ScalingConfiguration() : null)
                     .storageType(getStorageType())
-                    .allocatedStorage(getAllocatedStorage())
+                    .allocatedStorage(getAllocatedStorage() == null ? null : getAllocatedStorage())
```
Comment (harjain99) [inline src/main/java/gyro/aws/rds/DbClusterResource.java:972]: We should need a ternary operator here, `getAllocatedStorage()` will also return null if `allocated-storage` is null.

File: src/main/java/gyro/aws/rds/DbClusterResource.java [hunk 2]
```diff
@@ -1059,11 +1061,12 @@ protected void doUpdate(Resource config, Set<String> changedProperties) {
...
```
Comment (harjain99) [inline src/main/java/gyro/aws/rds/DbClusterResource.java:1069]: same as above

**Review by harjain99 — CHANGES_REQUESTED** [review]
Added requested changes

**Comment by Russsnizza** [comment]
Updated.

**Review by harjain99 — APPROVED** [review]

**Pull request merged**